src/main/java/com/synopsys/integration/alert/common/descriptor/config/context/DescriptorActionApi.java

```diff
@@ -41,13 +46,44 @@
 
 public abstract class DescriptorActionApi {
 
-    public abstract void validateConfig(final FieldModel fieldModel, final Map<String, String> fieldErrors);
+    public static final String REQUIRED_FIELD_MISSING = "Required field missing";
+
+    public void validateConfig(final Collection<ConfigField> descriptorFields, final FieldModel fieldModel, final Map<String, String> fieldErrors) {
+        for (final ConfigField field : descriptorFields) {
+            final String fieldKey = field.getKey();
+            final Optional<FieldValueModel> optionalField = fieldModel.getField(fieldKey);
+            if (field.isRequired()) {
+                if (optionalField.isEmpty()) {
```

Comment:
For this outer if-statement, I would be more comfortable doing an `.isPresent()` check and before a `.get()` in case the code ever changes. We should also minimize the number of times we do the get by storing the value from the Optional in a local variable.